base/linalg/generic.jl

```diff
@@ -704,6 +704,12 @@ function issymmetric(A::AbstractMatrix)
     if indsm != indsn
         return false
     end
+    if indsm == Base.OneTo(1)
+        if isnan(A[1,1])
+            return false
+        end
+        return true
+    end
```

Comment:
I don't think that this is sufficient although it will fix the `eig` issue. For larger matrices, we'd have to check that whole diagonal so instead the loop ranges below should be made one larger to include the diagonal.